Update ldap-group-constrained-team-channel.rst (#2913)
mattermost/docs: source/deployment/ldap-group-constrained-team-channel.rst

 - invitation through a mention
 - invitation through the /invite slash command 
 - added to the channel with “add members”

> Users, however, can remove themselves from teams and private channels managed by synchronized groups. 

Reviewer: ```suggestion
Users can remove themselves from teams and private channels managed by synchronized groups. 
```

2. Ensure **Team Settings > General > Allow any user with an account on this server to join this team** is set to ``No``. 
3. Convert the team to have its membership managed by synchronized groups by running the `group team enable CLI command <https://docs.mattermost.com/administration/command-line-tools.html#mattermost-group-team-enable>`_.

To manage membership of a private channel with synchronized groups: 

1. Navigate to **System Console > User Management > Channels**. Select the channel you want to manage with group synchronization.  
2. Under **Channel Management**, enable **Sync Group Members**. Please ensure the channel is set to ``private``. 
3. Add one or more groups to the channel. If there are groups already associated to default users into the channel, they will already be present.  
4. Review the notice in the footer of the screen for any users that are not part of groups who will be removed from the channel on the next synchronization.
5. Click save. Members will be updated on the next scheduled AD/LDAP synchronization. 

Alternatively you can use the CLI tool to set a private channel to be managed by groups: 

> 1. Ensure there is at least one group already associated to the channel. You can view and add default channels to a group via **System Console > User Management > Groups > Group Configuration**. Please see more information on adding default teams and channels `here <https://docs.mattermost.com/deployment/ldap-group-sync.html#add-default-teams-or-channels-for-the-group>`_. Additionally, you can use the CLI tool to view if there is already a group associated to the channel by running the `group channel list CLI command <https://docs.mattermost.com/administration/command-line-tools.html#mattermost-group-team-list>`_.

Reviewer 2: Wondering what happens if there isn't one group?

Author: For the CLI, I believe you will just be in a weird scenario where the policy is set, but there is no one in the channel/team because you didn't add any groups to it.  In the UI, we prevent you from changing the management to synced to groups without having a group.  @mkraft can you please confirm for the CLI?

Reviewer: The CLI prevents one from enabling group-constraints without at least one associated group (same as the system console web UI).

Author: Thanks for clarifying! Glad it is the same in both :)

**Does a team with its membership managed by groups have any effect on public channel access?**

Only users that are members of groups synchronized to team are able to discover and join public channels.  Private channels can also be managed by synchronized groups when a team is managed by synchronized groups. 

**Why don't users get readded to teams or channels once they have been removed from and then later re-added to the LDAP group?**

> The implementation of group removals does not currently differentiate between users who have removed themselves or have been removed by the LDAP synchronization process. Our design optimizes for users who have removed themselves from a team or channel.  In the future, we may add more tools to allow admins to re-add all users who have been removed and even prevent users from leaving a team or channel. Additionally, LDAP users who are not accessible to Mattermost based on filters will be removed from the groups and may be removed from teams and channels. If they were removed from teams and channels then they will not be re-added to those teams and channels upon becoming subsequently re-accessible to Mattermost.  

Reviewer 2: I would break into two paragraphs, starting with `Additionally...`

**Does a team with its membership managed by groups have any effect on public channel access?**

Only users that are members of groups synchronized to team are able to discover and join public channels.  Private channels can also be managed by synchronized groups when a team is managed by synchronized groups. 

**Why don't users get readded to teams or channels once they have been removed from and then later re-added to the LDAP group?**

> The implementation of group removals does not currently differentiate between users who have removed themselves or have been removed by the LDAP synchronization process. Our design optimizes for users who have removed themselves from a team or channel.  In the future, we may add more tools to allow admins to re-add all users who have been removed and even prevent users from leaving a team or channel. 

Reviewer: ```suggestion
The implementation of group removals does not currently differentiate between users who have removed themselves or have been removed by the LDAP synchronization process. Our design optimizes for users who have removed themselves from a team or channel.  In the future, we may add the ability for admins to re-add users who have been removed, and even prevent users from leaving, a team or channel. 
```

> Additionally, LDAP users who are not accessible to Mattermost based on filters will be removed from the groups and may be removed from teams and channels. If they were removed from teams and channels then they will not be re-added to those teams and channels upon becoming subsequently re-accessible to Mattermost.  

Reviewer: ```suggestion
Additionally, LDAP users who are not accessible to Mattermost based on filters will be removed from the groups and from group-synced teams and channels. If they were removed from teams and channels then they will not be re-added to those teams and channels upon becoming subsequently re-accessible to Mattermost.
```
